target/linux/ramips/mt7621/base-files/etc/init.d/bootcount

```diff
@@ -8,6 +8,9 @@ boot() {
 		[ -n "$(fw_printenv bootcount bootchanged 2>/dev/null)" ] &&\
 			echo -e "bootcount\nbootchanged\n" | /usr/sbin/fw_setenv -s -
 		;;
+	beeline,smartbox-turbo-plus)
+		echo -ne "$(echo FF | sed 's/../\\x&/g')" | dd of=/dev/mtdblock4 bs=1 seek=$((0x20001)) count=1
```

Comment:
This should be
```printf '\xff' | dd of=/dev/mtdblock4 bs=1 seek=$((0x20001)) count=1

Reply:
Better to do a check at the beginning.
```[ "$(dd if=/dev/mtd4 skip=$((0x20001)) count=1 bs=1 2>/dev/null)" = "$(printf '\xff')" ] || printf '\xff' | dd of=/dev/mtdblock4 seek=$((0x20001)) bs=1 count=1```

Reply:
What is the point of this check? And I see no pint in using printf to compare.
```
[[ $(dd if=/dev/mtd4 skip=$((0x20001)) count=1 bs=1 2>/dev/null) -eq 0xff ]]
```

Reply:
Check if the counter has not incremented, then there is nothing to do. Useful for those who changed the bootloader to another. 

Have you checked?
```
root@openwrt:~# [[ $(dd if=/dev/mtd4 skip=$((0x20001)) count=1 bs=1 2>/dev/null)
 -eq 0xff ]] && echo 0 || echo 1
ash: �: out of range
1
```

Reply:
This syntax is the last thing we need to worry about now. I'll fix it later.

Reply:
Fixed.